src/secret_sharing/mod.rs

```diff
@@ -1,5 +1,14 @@
 mod malicious_replicated;
 mod replicated;
 
+use crate::ff::Field;
 pub use malicious_replicated::MaliciousReplicated;
 pub use replicated::Replicated;
+use std::fmt::Debug;
+use std::ops::{Add, AddAssign, Mul, Neg, Sub, SubAssign};
+
+/// Secret share of a secret has additive and multiplicative properties.
+pub trait SecretShare<F>: Add + AddAssign + Neg + Sub + SubAssign + Mul<F> + Debug + Sized {}
```

Comment:
I would prefer to call it: `SecretSharing<F>` since that sounds more conversational.

Reply:
Can we somehow add "possible to get a sharing of ONE", (or really of any constant) to this?

Reply:
I could add something like `fn one(role: Identity) -> Self` but this does not seem to be enough to construct maliciously secure share? we may also have `fn one(ctx: ProtocolContext<_, Self, F)` but that would require r_share to be provided from the context and I don't know what index to use to generate randomness at this point.